Wrappers/Python/cil/framework/processors.py

```diff
@@ -0,0 +1,265 @@
+            raise TypeError("Input type mismatch: got {0} expecting {1}" \
+                            .format(type(dataset), DataContainer))
+
+
+    def check_input(self, dataset):
+        '''Checks parameters of the input DataContainer
+
+        Should raise an Error if the DataContainer does not match expectation, e.g.
+        if the expected input DataContainer is 3D and the Processor expects 2D.
+        '''
+        raise NotImplementedError('Implement basic checks for input DataContainer')
+
+    def get_output(self, out=None):
+        """
+        Runs the configured processor and returns the processed data
+
+        Parameters
+        ----------
+        out : DataContainer, optional
+           Fills the referenced DataContainer with the processed data and suppresses the return
+
+        Returns
+        -------
+        DataContainer
+            The processed data. Suppressed if `out` is passed
+        """
+        if self.output is None or self.shouldRun:
+            if out is None:
+                out = self.process()
+            else:
+                self.process(out=out)
+
+            if self.store_output:
+                self.output = out.copy()
+
+            return out
+
+        else:
+            return self.output.copy()
+
+
+    def set_input_processor(self, processor):
+        if issubclass(type(processor), DataProcessor):
+            self.__dict__['input'] =  weakref.ref(processor)
+        else:
+            raise TypeError("Input type mismatch: got {0} expecting {1}"\
+                            .format(type(processor), DataProcessor))
+
+    def get_input(self):
+        '''returns the input DataContainer
+
+        It is useful in the case the user has provided a DataProcessor as
+        input
+        '''
+        if self.input() is None:
+            raise ValueError("Input has been deallocated externally")
+        elif issubclass(type(self.input()), DataProcessor):
+            dsi = self.input().get_output()
+        else:
+            dsi = self.input()
+        return dsi
+
+    def process(self, out=None):
+        raise NotImplementedError('process must be implemented')
+
+    def __call__(self, x, out=None):
+
+        self.set_input(x)
+
+        if out is None:
+            out = self.get_output()
+        else:
+            self.get_output(out=out)
+
+        return out
+
+
+class DataProcessor(Processor):
+    '''Basically an alias of Processor Class'''
+    pass
+
+
+class DataProcessor23D(DataProcessor):
+    '''Regularizers DataProcessor
+    '''
+
+    def check_input(self, dataset):
+        '''Checks number of dimensions input DataContainer
+
+        Expected input is 2D or 3D
+        '''
+        if dataset.number_of_dimensions == 2 or \
+           dataset.number_of_dimensions == 3:
+               return True
+        else:
+            raise ValueError("Expected input dimensions is 2 or 3, got {0}"\
+                             .format(dataset.number_of_dimensions))
+
+
```

Comment:
This class is entirely unused - begone!!!!

Reply:
I agree